Kotlin/src/main/kotlin/com/gildedrose/updaters/BackstagePassUpdater.kt

```diff
@@ -0,0 +1,20 @@
+package com.gildedrose.updaters
+
+import com.gildedrose.Item
+
+class BackstagePassUpdater: BaseUpdater() {
+
+    override fun update(item: Item) {
+        item.updateQuality()
+        cleanUp(item)
+    }
+
+    private fun Item.updateQuality() {
+        when {
```

Comment:
I really like this method! It's a very elegant way to update the quality.

I'd recommend using some more functional principles to neaten it up a little, along the lines of:
```
    quality += when {
        sellIn <= 0 -> -quality
        sellIn <= 5 -> 3
        sellIn <= 10 -> 2
        else -> 1
    }
```

There are a couple of other places that this might be useful too :)